### DISCUSSION TITLE
Getting the local time of a user (Different Timezones)

### DISCUSSION THREAD

**Discussion by samjesus8** [DSharpPlus/DSharpPlus#1444] [Support] [answered]
# Summary of issue

I have this current command called "time" which is supposed to get the local time of the user who calls it. My aim is to get it working with users from different countries and I can't seem to find any property that gets the local time of the user. So far I am only able to use the timestamp property to get the time but it only gets the time of the bot which is set in the UK

# The command code
```
        [Command("time")]
        public async Task GetLocalTime(CommandContext ctx) 
        {
            string localTime = ctx.Message.Timestamp.ToString();

            var msg = new DiscordMessageBuilder()
                .AddEmbed(new DiscordEmbedBuilder()

                .WithColor(DiscordColor.Azure)
                .WithTitle("Current time for " + ctx.User.Username)
                .WithDescription(localTime)
                );

            await ctx.Channel.SendMessageAsync(msg);
        }
```

# What I am trying to achieve

1. Get the local time of the user who called the command
2. Display it in embed message and send it off to the channel where its called

**Comment by VelvetToroyashi** [comment, marked as answer]
Discord doesn't directly expose user's time zones as that could be considered a breach of privacy. If you want a user-agnostic way to display local time, consider using `Formatter.Timestamp`, which uses Discord's built-in timestamp markdown.

**Reply by samjesus8** [reply]
So there's no way to get the user's local time from "ctx", because if i use "DateTime.Now" that's just going to do the same thing

**Reply by VelvetToroyashi** [reply]
As I said, you should use the timestamp markdown, which is displayed in the user's local time anyway

**Reply by VelvetToroyashi** [reply]
![image](https://user-images.githubusercontent.com/42438262/210257868-d2707e57-652d-43aa-a9ad-c2e88f5ac5ca.png)
